src/trackByFunctionRule.ts

```diff
@@ -0,0 +1,69 @@
+import * as Lint from 'tslint';
+import * as ts from 'typescript';
+import { NgWalker } from './angular/ngWalker';
+import * as ast from '@angular/compiler';
+import { BasicTemplateAstVisitor } from './angular/templates/basicTemplateAstVisitor';
+
+export class Rule extends Lint.Rules.AbstractRule {
+  public static metadata: Lint.IRuleMetadata = {
+    ruleName: 'trackBy-function',
+    type: 'functionality',
+    description: `Ensures a TrackBy function is used.`,
+    rationale: `Using TrackBy is considired as a best pratice.`,
+    options: null,
+    optionsDescription: `Not configurable.`,
+    typescriptOnly: true
+  };
+
+  public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+    return this.applyWithWalker(
+      new NgWalker(sourceFile,
+        this.getOptions(), {
+          templateVisitorCtrl: TrackByTemplateVisitor,
+        }));
+  }
+}
+
+const noTrackByRe = new RegExp('trackBy:');
```

Comment:
`/trackBy\s*:/` might be a better regex because of the whitespace which is ignored by default